plugins/web/opentelemetry-instrumentation-long-task/src/instrumentation.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hrTime } from '@opentelemetry/core';
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+import { VERSION } from './version';
+
+// Currently missing in typescript DOM definitions
+interface PerformanceLongTaskTiming extends PerformanceEntry {
+  attribution: TaskAttributionTiming[];
+}
+
+interface TaskAttributionTiming extends PerformanceEntry {
+  containerType: string;
+  containerSrc: string;
+  containerId: string;
+  containerName: string;
+}
+
+const LONGTASK_PERFORMANCE_TYPE = 'longtask';
+
+export class LongTaskInstrumentation extends InstrumentationBase {
+  readonly component: string = 'long-task';
+  readonly version: string = VERSION;
+  moduleName = this.component;
+
+  private _observer?: PerformanceObserver;
+
+  /**
+   *
+   * @param config
+   */
+  constructor(config: InstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-long-task', VERSION, config);
+  }
+
+  init() {}
+
+  private isSupported() {
+    if (
+      typeof PerformanceObserver === 'undefined' ||
+      !PerformanceObserver.supportedEntryTypes
+    ) {
+      return false;
+    }
+
+    return PerformanceObserver.supportedEntryTypes.includes(
+      LONGTASK_PERFORMANCE_TYPE
+    );
+  }
+
+  private _createSpanFromEntry(entry: PerformanceLongTaskTiming) {
+    const span = this.tracer.startSpan(LONGTASK_PERFORMANCE_TYPE, {
+      startTime: hrTime(entry.startTime),
+    });
+    span.setAttribute('component', this.component);
+    span.setAttribute('longtask.name', entry.name);
+    span.setAttribute('longtask.entry_type', entry.entryType);
+    span.setAttribute('longtask.duration', entry.duration);
```

Comment:
I assume there is no spec for these attributes. Can we see if there is any guidance from the spec/TC on how we should handle custom or non-specified attributes? If these are ever specified in the future in a different format it could make us non-compliant.

Reply:
@dyladan instrumentation are not stable yet so this shouldn't be considered a breaking change if in the future they are in the spec ?

Reply:
No but in general I think we should try to be consistent.